Backend/Middleware/RequestMiddleware.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Backend.Models.DTO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Backend.Middleware
+{
+    public class RequestMiddleware
+    {
+        private const string ApiHeader = "X-Api-Request";
+
+        private readonly RequestDelegate _next;
+
+        public RequestMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            httpContext.Request.Headers.TryGetValue(ApiHeader, out var headers);
+            if (!IsHeaderLegit(headers) || headers != "true")
+            {
+                var error = new ErrorDTO
+                {
+                    Type = 400,
+                    Title = "Invalid Headers!",
+                    Details = null
+                };
+                httpContext.Response.StatusCode = 400;
+                var jsonString = JsonConvert.SerializeObject(error, Formatting.Indented);
```

Comment:
Since we already know this variable will be `string`, name `jsonString` is redundant according to Uncle Bob. Also, since this is a primitive type, we can use `string` instead of `var`.

Reply:
Changed :)